### PR TITLE
introduce specific interfaces for local and session storage

### PR DESCRIPTION
**Pull request by celluj34**

### CODE REVIEW

**Comment by celluj34** [comment]
Looks like the CI server doesn't have the latest .Net Core 3 SDK installed.

**Comment by celluj34** [comment]
@galvesribeiro Any chance you could review these? I would like to use these interfaces in my own code for unit testing.

**Comment by galvesribeiro** [comment]
Hey! Sorry for the delay. Please fix the comments on the review and we should be good to merge it.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     - name: Setup .NET Core
       uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: 3.1.100-preview1-014459
+        dotnet-version: 3.1.0-preview3.19555.2
```
Comment (galvesribeiro) [inline .github/workflows/publish.yml]: It should be `3.1.100-preview3-014645`. What you added here is the package Version while this entry requires the SDK version. Please update it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,20 +0,0 @@
-name: CI
-
```
Comment (galvesribeiro) [inline .github/workflows/ci.yml]: Why this file was deleted? It should be there and it should have the same `3.1.100-preview3-014645` SDK version.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (galvesribeiro) [reply .github/workflows/publish.yml]: Thanks

File: .github/workflows/ci.yml (hunk 2 above)
Comment (celluj34) [reply .github/workflows/ci.yml]: Sorry, was having trouble doing it on mobile. I'll get it all fixed

Comment (galvesribeiro) [reply .github/workflows/ci.yml]: No rush, take your time.

**Pull request merged**